packages/nimble-components/build/generate-workers/source/matrix-renderer.ts

```diff
@@ -8,16 +9,172 @@ import { expose } from 'comlink';
+                prev = xIndex
+            }
+        }
+        this.scaledColumnIndex = Float64Array.from(scaledColumnIndex);
+        this.columnIndexPositions = Int32Array.from(columnPositions);
+    }
+
+    public setRowIndexes(rowIndexesBuffer: Int32Array): void {
+        this.rowIndexes = rowIndexesBuffer;
+        this.scaledRowIndex = new Float64Array(this.rowIndexes.length);
+        for (let i = 0; i < this.rowIndexes.length; i++) {
+            this.scaledRowIndex[i] = this.calculateYScaledIndex(this.rowIndexes[i]!);
+        }
+    }
+
+    public setMargin(margin: { top: number, right: number, bottom: number, left: number }): void {
+        this.margin = margin;
+    }
+
+    public setCanvasCorners(topLeft: { x: number, y: number }, bottomRight: { x: number, y: number }): void {
+        this.topLeftCanvasCorner = topLeft;
+        this.bottomRightCanvasCorner = bottomRight;
+    }
+
+    public setDiesDimensions(data: Dimensions): void {
+        this.dieDimensions = { width: data.width, height: data.height };
+    }
+
+    public setScaling(scaleX: number, scaleY: number): void {
+        this.scaleX = scaleX;
+        this.scaleY = scaleY;
+    }
+
+    public setBases(baseX: number, baseY: number): void {
+        this.baseX = baseX;
+        this.baseY = baseY;
+    }
+
+    public setTransform(transform: Transform): void {
+        this.transform = transform;
+    }
+
+    public setCanvas(canvas: OffscreenCanvas): void {
+        this.canvas = canvas;
+        this.context = canvas.getContext('2d')!;
+    }
+
+    public getMatrix(): WaferMapTypedMatrix {
+        return {
+            columnIndexes: this.columnIndexes,
+            rowIndexes: this.rowIndexes,
+            values: this.values
+        };
+    }
 
     public emptyMatrix(): void {
-        this.dieMatrix = Uint8Array.from([]);;
+        this.columnIndexes = Int32Array.from([]);
+        this.rowIndexes = Int32Array.from([]);
+        this.values = Float64Array.from([]);
+    }
+
+    public scaleCanvas(): void {
+        this.context.translate(
+            this.transform.x,
+            this.transform.y
+        );
+        this.context.scale(
+            this.transform.k,
+            this.transform.k
+        );
     }
 
     public updateMatrix(
-        data: Iterable<number>
+        data: WaferMapMatrix
     ): void {
-        this.dieMatrix = Uint8Array.from(data);
+        this.columnIndexes = Int32Array.from(data.columnIndexes);
+        this.rowIndexes = Int32Array.from(data.rowIndexes);
+        this.values = Float64Array.from(data.values);
+    }
+
+    public setCanvasDimensions(data: Dimensions): void {
+        this.canvas.width = data.width;
+        this.canvas.height = data.height;
+    }
+
+    public clearCanvas(): void {
+        this.context.clearRect(
+            0,
+            0,
+            this.canvas.width,
+            this.canvas.height
+        );
+    }
+
+    public drawWafer(): void {
```

Comment:
@Razvan1928 is this implementation being verified through existing tests?

Reply:
Now that you are saying I added only one test for this pr, will add a few more

Reply:
Added 4 more tests, can be found in `matrix-renderer.spec.ts`

Reply:
@Razvan1928, I'm unresolving this just to get some clarity. While I appreciate the added tests, I'm still curious if the `drawWafer` method (particular the one for the worker-renderer) is being tested anywhere.

Reply:
After talking with Razvan, this implementation will be verified through Chromatic tests.